packages/mds-agency-cache/index.ts

```diff
@@ -202,7 +181,7 @@ async function writeEvent(event: VehicleEvent) {
   // FIXME cope with out-of-order -- check timestamp
   // logger.info('redis write event', event.device_id)
   try {
-    if (event.event_type === 'deregister') {
+    if (tail(event.event_types) === 'decommissioned') {
```

Comment:
What happens if `decomissioned` is in one of the `event_types`, but just not the last one?

Reply:
It seems unlikely that a decommissioned vehicle would have an event with an array of event types where `decommissioned` was in the middle. And if it does, say it has an event with `event_types` `['decommissioned', 'trip_start']`, something has gone wrong, but we probably shouldn't be decommissioning it, since it's still going on trips.